app/Middleware/Queue/SubmissionService.php

```diff
@@ -23,6 +23,60 @@
 use CDash\Log;
 use CDash\Middleware\Queue;
 
+/**
+ * Class SubmissionService
+ * @package CDash\Middleware\Queue
+ *
+ * Usage:
+ *
+ * The queue package, Bernard, used by CDash has the notion of consumers of queues and producers
+ * of messages for queues. SubmissionService is an implementation of a Bernard consumer that
+ * also statically creates a message for a queue regarding a CTest submission in the format that it
+ * expects (via SubmissionService::createMessage).
+ *
+ * Bernard has a naming convention that must be followed for consumption of queues. Queue names
+ * have no spaces, and the first letter of each word in the name of the queue is capitalized. For
+ * instance, in the general case of the SubmissionClass, one needs to create a queue named do-submit
+ * which will result in the Bernard based queue name being DoSubmit, which, not coincidentally,
+ * has the affect that when consuming a message, the consumer that has been registered with Bernard
+ * (in this case SubmissionService) will call a method on that service named doSubmit. In summary:
+ *
+ *   1. If the name of the queue is do-submit
+ *   2. The name provided to the message must be DoSubmit
+ *   3. This requires that the consumer associated with the message have a method named doSubmit
+ *
+ * If you wish to provide your own queue name, for instance, drake-cdash, you would use this
+ *  class in the following way:
+ *
+ *   1. Ensure that your configuration of the queue is correct. The queue configuration file is
+ *      located at <cdash root>/config/queue.php.
+ *
+ *   2. Create a queue
+ *      $queue = new Queue();
+ *
+ *   3. Register a consumer, this, with the queue
+ *      $queue_name = 'DrakeCdash'; // Bernard naming convention for queue named drake-cdash
+ *      $service = new SubmissionService($queue_name); // Notice we provide the optional name arg
+ *      $queue->addService($queue_name, $service);
+ *
+ *   4. Create a message
+ *      // required arguments
+ *      $arguments_to_create_message = [
+ *        'file' => </path/to/file/being/submitted/by/ctest>,
+ *        'project' => <The name of the project of whom the file belongs to>,
+ *        'md5' => <A hash of the file created by md5_file('/path/to/file')>,
+ *        'checksum' => <A boolean indicating whether or not to create a checksum for the file>,
+ *      ];
+ *
+ *      // optionally, to change the name of the queue used by SubmissionService
+ *      $arguments_to_create_message['queue_name'] = $queue_name;
+ *
+ *      $message = SubmissionService::createMessage($arguments_to_create_message);
+ *
+ *   5. Produce the message (send it to the queue)
+ *      $queue->produce($message);
+ *
+ */
```

Comment:
Thanks for thoroughly documenting all the ins and outs of naming your own queue.